src/ConversationData.res

```diff
@@ -62,97 +62,45 @@ and attachment = {
-      latest_message: json |> field("latest_message", single_message),
-      rating: json
-      |> optional(field("rating", string))
-      |> (
-        rating =>
-          switch rating {
-          | Some("green") => Green
-          | Some("yellow") => Yellow
-          | Some("red") => Red
-          | _ => Unrated
-          }
-      ),
-      has_attachments: json |> field("has_attachments", bool),
-      notes: json |> field("notes", string),
-      is_read: json |> field("is_read", bool),
-      is_ignored: json |> field("is_ignored", bool),
-      is_replied_to: json |> field("is_replied_to", bool),
-      has_been_replied_to: json |> field("has_been_replied_to", bool),
-      is_in_trash: json |> field("is_in_trash", bool),
-    }
-  }
-  and single_message = (json): message => {
-    open Json.Decode
-    {
-      id: json |> field("id", int),
-      /* conversation: json |> field("conversation", conversation), */
-      conversation_id: json |> field("conversation_id", int),
-      type_: json
-      |> field("type", string)
-      |> (
-        type_ =>
-          switch type_ {
-          | "incoming" => Incoming
-          | "outgoing" => Outgoing
-          | _ => failwith("invalid message type")
-          }
-      ),
-      content: json |> field("content", string),
-      date: json |> field("date", string),
-      attachments: json |> field("attachments", many_attachments),
-    }
-  }
-  and single_attachment = (json): attachment => {
-    open Json.Decode
-    {
-      filename: json |> field("filename", string),
-      mimetype: json |> field("mimetype", string),
-      url: json |> field("url", string),
-    }
-  }
-  and many_conversations = json => json |> Json.Decode.array(single_conversation)
-  and many_messages = (json): array<message> => json |> Json.Decode.array(single_message)
-  and many_attachments = (json): array<attachment> => json |> Json.Decode.array(single_attachment)
+  /** Unsafely coerces a backend json payload for a conversation. */
+  external single_conversation: Js.Json.t => conversation = "%identity"
+
+  /** Unsafely coerces a backend json payload for an array of conversations. */
+  external many_conversations: Js.Json.t => array<conversation> = "%identity"
+
+  /** Unsafely coerces a backend json payload for a message. */
+  external single_message: Js.Json.t => message = "%identity"
+
+  /** Unsafely coerces a backend json payload for an array of messages. */
+  external many_messages: Js.Json.t => array<message> = "%identity"
 }
 
-let fetchConversations = (immobilie_id): Js.Promise.t<array<conversation>> => {
-  open Promise2
-  Fetch.fetchWithInit(
-    apiBaseUrl ++ ("/anfragen/immobilie/" ++ (string_of_int(immobilie_id) ++ "/conversations")),
-    Fetch.RequestInit.make(~credentials=Include, ()),
-  )
-  ->then(Fetch.Response.json)
-  ->then(json => json->Decode.many_conversations->resolve)
+let fetchConversations = async (immobilie_id): array<conversation> => {
```

Comment:
We now have async/await available!

ReScript's async/await is a little different, since the return type of the function is not a `promise<conversation>` like in TS, but rather just `conversation`.

The `promise` type is also new, which is just an alias to the original `Js.Promise.t` type.

More on the async/await feature can be found in the manual: https://rescript-lang.org/docs/manual/latest/async-await